integration-tests/src/xcm_rate_limiter.rs

```diff
@@ -0,0 +1,253 @@
+#![cfg(test)]
+
+use crate::polkadot_test_net::*;
+
+use frame_support::{assert_ok, pallet_prelude::Weight};
+use orml_traits::currency::MultiCurrency;
+use pallet_asset_registry::AssetType;
+use polkadot_xcm::prelude::*;
+use xcm_emulator::TestExt;
+
+/// Returns the message hash in the `XcmpMessageSent` event at the `n`th last event (1-indexed, so if the second to last
+/// event has the hash, pass `2`);
+fn get_message_hash_from_event(n: usize) -> Option<[u8; 32]> {
+	use cumulus_pallet_xcmp_queue::Event;
+	use hydradx_runtime::RuntimeEvent;
+	let RuntimeEvent::XcmpQueue(Event::XcmpMessageSent { message_hash }) = &last_hydra_events(n)[0] else {
+		panic!("expecting to find message sent event");
+	};
+	*message_hash
+}
+
+#[test]
+fn xcm_rate_limiter_should_limit_aca_when_limit_is_exceeded() {
```

Comment:
I would probably also test clearing of the limits in similar fashion after blocks pass